hedera/tutorials/smart-contracts/hss-x-evm-part-1-schedule-smart-contract-calls.mdx

```diff
@@ -0,0 +1,851 @@
+  console.log("Deploying contract with the account:", deployer.address);
+
+  // Deploy the AlarmClockSimple contract with initial HBAR funding
+  const AlarmClockSimple = await ethers.getContractFactory(
+    "AlarmClockSimple",
+    deployer
+  );
+
+  // Send 50 HBAR to the contract during deployment
+  // This HBAR will be used to pay for scheduled alarm executions
+  const HBAR_TO_SEND = "50";
+  console.log(
+    `Deploying with ${HBAR_TO_SEND} HBAR to fund scheduled executions... `
+  );
+
+  const contract = await AlarmClockSimple.deploy({
+    value: ethers.parseEther(HBAR_TO_SEND)
+  });
+  await contract.waitForDeployment();
+
+  const contractAddress = await contract.getAddress();
+  console.log("AlarmClockSimple contract deployed at:", contractAddress);
+
+  // Get the contract balance to verify funding
+  const balance = await ethers.provider.getBalance(contractAddress);
+  console.log("Contract HBAR balance:", ethers.formatEther(balance), "HBAR");
+}
+
+main().catch(console.error);
+```
+
+<Note>
+**Critical:  Why Does the Contract Need HBAR?**
+
+Every time a scheduled transaction executes (when the network automatically calls `triggerAlarm`), gas fees must be paid. Since the contract itself is the payer for these scheduled executions, it **must hold HBAR**.
+
+**For recurring alarms:**
+
+- Each `triggerAlarm` execution consumes gas
+- Each `triggerAlarm` then schedules the next alarm (more gas)
+- If the contract runs out of HBAR, scheduled executions will fail and recurring alarms will stop
+
+**Best practice:** Fund the contract with enough HBAR based on:
+
+- Your `SCHEDULED_CALL_GAS_LIMIT` (set to 2,000,000 in this example)
+- Expected number of alarm executions
+- For recurring alarms, estimate how long you want them to run
+
+You can always add more HBAR by sending a transaction to the contract address.
+
+</Note>
+
+Deploy your contract by executing the script:
+
+```bash
+npx hardhat run scripts/deploy.ts --network testnet
+```
+
+<Check>Copy the deployed address—you'll need this in subsequent steps. </Check>
+
+The output looks like this:
+
+```bash
+Deploying contract with the account: 0xA98556A4deeB07f21f8a66093989078eF86faa30
+Deploying with 5 HBAR to fund scheduled executions...
+AlarmClockSimple contract deployed at: 0x76CFE47809A84e62D633562c350e775e0dB7E1bF
+Contract HBAR balance: 5.0 HBAR
+```
+
+---
+
+## Step 4: Set a One-Shot Alarm
+
+Create a script (`setOneShot.ts`) in your `scripts` directory to set a one-shot alarm. Don't forget to replace the `<your-contract-address>` with the address you've just copied.
+
+```typescript scripts/setOneShot.ts
+import { network } from "hardhat";
+
+const { ethers } = await network.connect({ network: "testnet" });
+
+async function main() {
+  const [signer] = await ethers.getSigners();
+  console.log("Using signer:", signer.address);
+
+  const contractAddress = "0x76CFE47809A84e62D633562c350e775e0dB7E1bF"; // Replace with your deployed contract address
```

Comment:
Same here to replace with demo value

Reply:
Can we maybe also use the `CONTRACT_ADDRESS=<your address>` solution here? I find it much easier that I don't have to think about replacing a value in the script?